bench/src/benchmark_result.rs

```diff
@@ -24,21 +25,61 @@ impl From<Vec<BenchmarkResult>> for BenchmarkResults {
         Self { results }
     }
 }
+#[derive(Debug, Clone, PartialEq)]
+struct BenchStatistics {
+    total_throughput: f64,
+    messages_per_second: f64,
+    average_latency: f64,
+    average_throughput: f64,
+    total_duration: f64,
+}
 
-impl Display for BenchmarkResults {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        let total_size_bytes = self.results.iter().map(|r| r.total_size_bytes).sum::<u64>();
+impl BenchmarkResults {
+    fn get_test_type(&self) -> Option<BenchmarkKind> {
```

Comment:
Make is Result and return some Error on (false, false) branch

Reply:
Done